aikau/src/main/resources/alfresco/services/actions/CopyMoveService.js

```diff
@@ -201,6 +234,42 @@ define(["dojo/_base/declare",
                }
             }));
          }
+      },
+
+       /**
+        * Handles the actual create link XHR call
+        *
+        * @instance
+        * @param {boolean} copy A boolean indicating if this is a copy action or not
+        * @param {object} payload The payload from confirmation of the action dialog
+        */
+      onCreateLinkLocationSelected: function alfresco_services_ActionService__onCreateLinkLocationSelected(payload) {
+          this.alfUnsubscribeSaveHandles([this._actionCopyHandle]);
+
+          // Get the locations to create link to and the documents to them...
+          if (payload.dialogContent)
+          {
+             when(payload.dialogContent, lang.hitch(this, function(content) {
```

Comment:
Because this block is so similar to the one in `onCopyMoveLocationSelected` we could actually have "hitched" the same function on both subscriptions with an additional argument indicating the function to call if validation checks pass (e.g. `performCreateLinkAction` or `performAction`)